### ISSUE TITLE
MSTest project error in Visual Studio 2019 ver. 16.4.4 or 16.4.5 - Unable to determine the provider name for provider factory of type 'System.Data.SQLite.SQLiteFactory'.

### ISSUE THREAD

**Issue by dcppat** [ErikEJ/SqlCeToolbox#837]
While trying to run a UnitTest from a MSTest project I get the following message when executing the script: var cv = (from c in db.Cv select c).ToList();

**"Unable to determine the provider name for provider factory of type 'System.Data.SQLite.SQLiteFactory'. Make sure that the ADO.NET provider is installed or registered in the application config.**

If I run the method directly (in debug mode - not calling it from the MSTest project) is working just fine.

Also, the same solution (project) works just fine in VS.NET 2017 ver. 15.9.4

### Steps to reproduce
I followed the steps described on the Wiki

Further technical details

Toolbox/Power Tools version: (found in About dialog - blue questionmark icon)

Version 4.7.667.0 - 1,490,146 downloads

SQL Server Compact 4.0 in GAC - No
SQL Server Compact 4.0 DbProvider - No

SQL Server Compact 4.0 DDEX provider - No
SQL Server Compact 4.0 Simple DDEX provider - Yes

SQL Server Compact 3.5 in GAC - No
SQL Server Compact 3.5 DbProvider - No

SQL Server Compact 3.5 DDEX provider - No

Sync Framework 2.1 SqlCe 3.5 provider - No

SQLite ADO.NET Provider included: 1.0.109.0
SQLite EF6 DbProvider in GAC - Yes
System.Data.SQLite DDEX provider - No
SQLite Simple DDEX provider - Yes

Database engine: SQlite

Visual Studio Visual Studio 2019 Version 16.4.5

**Comment by ErikEJ** [comment]
This issue has very little to do with the Toolbox, but in order for me to help in any way, I need a repro project from you.

**Comment by dcppat** [comment]
Erik,

Please, find attached a repro solution. It has three layers. UI - user interface, DA- data access and UnitTest. DB attached as well.

Running the UI works just fine. 

The error occurs by debugging or running the UnitTest in VS 2019. Works perfectly in VS 2017

Thanks in advance

[AID.zip](https://github.com/ErikEJ/SqlCeToolbox/files/4264992/AID.zip)

**Comment by GrooverFromHolland** [comment]
I Had this problem to. Mi solution was to add these lines in App.config at  <DbProviderFactories>:
` <remove invariant="System.Data.SQLite" />
      <add name="SQLite Data Provider" invariant="System.Data.SQLite"
           description=".NET Framework Data Provider for SQLite" 
           type="System.Data.SQLite.SQLiteFactory, System.Data.SQLite" />`

**Comment by dcppat** [comment]
Thanks Groover. This is the first thing that I tried, but it didn't make a difference.

I was able to solve it by upgrading the following NuGetPackages for the MSTest project.
  1) package id="EntityFramework" version="6.4.0" targetFramework="net461"
  2) package id="MSTest.TestAdapter" version="2.1.0" targetFramework="net461"
  3) package id="MSTest.TestFramework" version="2.1.0" targetFramework="net461"
